coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1687,36 +1687,88 @@ def pad(context, node):
     context.add(res)
 
 
+@register_torch_op
+def adaptive_avg_pool1d(context, node):
+    _adaptive_pool1d(context, node, mb.reduce_mean)
+
+
 @register_torch_op
 def adaptive_avg_pool2d(context, node):
     _adaptive_pool2d(context, node, mb.avg_pool, mb.reduce_mean)
 
 
+def _adaptive_pool1d(context, node, reduce_op):
+    inputs = _get_inputs(context, node, expected=2)
+    x = inputs[0]
+    assert len(inputs[1].val) == 1
+    out_length = inputs[1].val[0]
+
+    if len(x.shape) == 3:
+        # 3D input
+        begin_prefix = [0, 0]
+        end_prefix = [x.shape[0], x.shape[1]]
+        out_shape = [x.shape[0], x.shape[1], out_length]
+    else:
+        # 2D input
+        assert len(x.shape) == 2
+        begin_prefix = [0]
+        end_prefix = [x.shape[0]]
+        out_shape = [x.shape[0], out_length]
+
+    pool_results = []
+    for start, end in _get_kernel_indexes_1d_for_adaptive_pooling(x.shape[-1], out_length):
+        cur_kernel = mb.slice_by_index(
+            x=x,
+            begin=begin_prefix+[start],
```

Comment:
just curious,
is this a correct syntax or `begin=begin_prefix + [start],`?

Reply:
Your syntax is correct; there should be a space on either side of the plus sign. I'll change it.

Reply:
changed